### ISSUE TITLE
Let's get configurations and home-manager working with stable and unstable

### ISSUE THREAD

**Issue by voidzero** [Misterio77/nix-starter-configs#15]
Hi again,

Now that 22.11 is stable, I have tried to update my nix flakes to 22.11, and I synchronized my config files to the most recent version of this repo. This went without much hassle, but now again I am trying to add nix-unstable to the mix and am failing somewhere, because my changes *seem* to work, but for some reason the allowUnfree does not get accepted.

What I did so far was as follows. First, my `flake.nix` file contains this:

```
 inputs = {
    # Nixpkgs
    hardware.url = "github:nixos/nixos-hardware";
    nixpkgs.url  = "github:nixos/nixpkgs/nixos-22.11";
    nixpkgs-unstable.url = "github:nixos/nixpkgs/nixos-unstable";

    # Home manager
    home-manager.url = "github:nix-community/home-manager/release-22.11";
    home-manager.inputs.nixpkgs.follows = "nixpkgs";
  };

outputs = { self, nixpkgs, nixpkgs-unstable, home-manager, ... }@inputs:

```
in `extraSpecialArgs` under my username, well, let me just post the entire stanza:

```
        "mark@myhost" = home-manager.lib.homeManagerConfiguration {
          pkgs = nixpkgs.legacyPackages.x86_64-linux; # Home-manager requires 'pkgs' instance
          extraSpecialArgs = {
            inherit inputs outputs overlays;
            pkgs-unstable = nixpkgs-unstable.legacyPackages.x86_64-linux;
          };
          modules = [
            ./home-manager/myhost-mark.nix
          ];
        };

```
And finally in my `home-manager/myhost-mark.nix`, first: 
```
{ inputs, outputs, lib, config, pkgs, pkgs-unstable, ... }: {
```

and then under `home.packages = with pkgs; [`
`pkgs-unstable.vivaldi`

Here's the problem: `error: Package ‘vivaldi-5.6.2867.36’ in /nix/store/v3h6hhxf4f5l5ix3p70jap046zl2plm1-source/pkgs/applications/networking/browsers/vivaldi/default.nix:102 has an unfree license (‘unfree’), refusing to evaluate.`

In other words: the `allowUnfree` is apparently not read for `nixpkgs-unstable`.

I got so close! But one issue is remaining: no matter what I try to do - add `allowUnfree = true` to `nixpkgs.nix`, or to `home-manager/myhost-mark.nix`, or add `allowUnfreePredicate = (_: true);` to `home-manager/myhost-mark.nix`... vivaldi-unstable will not install and keeps giving me this error. Please advise what to do :sweat_smile: 

Thank youuu <3
Mark

**Comment by Misterio77** [comment]
Hey

The idea is that both the nixpkgs-related nixos options affect the "main" nixpkgs instance only. If you want to `allowUnfree`, you gotta set it while instantiating your pkgs-unstable. This should do the trick:

```nix
{
  "mark@myhost" = home-manager.lib.homeManagerConfiguration {
    pkgs = nixpkgs.legacyPackages.x86_64-linux; # Home-manager requires 'pkgs' instance
    extraSpecialArgs = {
      inherit inputs outputs overlays;
      pkgs-unstable = import nixpkgs-unstable { system = "x86_64-linux"; config.allowUnfree = true; };
    };
    modules = [
      ./home-manager/myhost-mark.nix
    ];
  };
}

```

**Comment by voidzero** [comment]
Ah! That did the trick indeed!
![image](https://user-images.githubusercontent.com/1332504/210066603-6ecf65f6-c298-48eb-9833-4a8397327ed8.png)
Thank you! Now I can use both stable and unstable packages with home-manager. Very useful!
